book/src/using_rusty/build_description_file.md

```diff
@@ -42,51 +49,118 @@ To link several executables `rustyc` has the option to add libraries and automat
 ]
 ```
 
-## output
-
-Similarly to specifying an output file via the `-o` or `--output` option, in the build file we use `"output" : "output.so"` to define the output file. The default location is likewise to the location for the build file, namely the current directory. 
 
+### output
 
+Similarly to specifying an output file via the `-o` or `--output` option using the command line, in the build file we use `"output" : "output.so"` to define the output file. The default location is the current build directory. (see [Build Location](#build-location)) 
 
-## Optional Keys
 
 ### compile_type
-There are six options for choosing the `compile_type`. The valid options are:
+
+The following options can be used for the `compile_type`:
 <!-- TODO we should probably describe what each of those options do -->
-- `Static` bindings have to be done at compile time
-- `PIC` Position Independent Code
-- `Shared` (dynamic) binginds will be done dynamically
-- `Relocatable` generates Relocatable 
+- `Static` specifies that linking/binding must be done at compile time
+- `Shared` (dynamic) specifies that linking/bingind must be done dynamically (at runtime)
+- `PIC` Position Independent Code (Choosing this option implies that the linking will be done dynamically)
+- `Relocatable` generates relocatable object code (for combining with other object code)
 - `Bitcode` adds bitcode alongside machine code in executable file
-- `IR` Intermediate Representation
+- `IR` intermediate `llvm` representation
+
+The compile format is specified in the build description file as follows:  `"compile_type" : "Shared"`.
+The `compile_type` keyword is optional.
+
 
-To specify which of the above mentioned compile formats is wanted, it needs to be added to the build description file as followed: `"compile_type" : "Shared"`.
+### package_commands
+The `package_commands` keyword is optional.
+
+> TODO
+
+### Example
 
-# Example
 ```json
 {
     "files" : [
         "examples/hw.st",
         "examples/hello_world.st",
-        "examples/ExternalFunctions.st"
+        "examples/ExternalFunctions.st",
+        "examples/*.dt"
     ],
     "compile_type" : "Shared",
     "output" : "proj.so",
     "libraries" : [
         {
             "name" : "iec61131std",
             "path" : "path/to/lib",
+            "package" : "Copy",
             "include_path" : [
-                "examples/hw.st"
+                "examples/lib.st"
             ]
         },
         {
             "name" : "other_lib",
             "path" : "path/to/lib",
+            "package" : "System",
             "include_path" : [
                 "examples/hello_world.st"
             ]
         }
     ]
 }
 ```
+
+## Build Parameters
+
+The `build` subcommand exposes the following optional parameters: 
+
+### `--build-location`
+
+The build location is the location all build files will be copied to. </br>
+By default the build location is the `build` folder in the root of the project.</br>
```

Comment:
is the root folder the working-directory of my shell? or where the build.json is?

Reply:
Where the build json is, i think i said that later, if not I'll append it here